packages/sdk/src/oracle/client.ts

```diff
@@ -176,20 +170,21 @@ export function factory(config: state.PartialConfig, emit: Emit): Client {
 
   // this first state machine is for user actions
   const sm = new StateMachine(store);
-  // this one is system actions used for long running commands independent of the user
-  const poller = new StateMachine(store);
 
   // start the request list checkers
   for (const [chainId, config] of Object.entries(fullConfig.chains)) {
-    poller.types.fetchPastEvents.create({ chainId: Number(chainId), startBlock: config.earliestBlockNumber }, "poller");
+    sm.types.fetchPastEvents.create(
```

Comment:
this is the command that was optimized, you can now configure the client to specify a max range to query and a block number threshold to ignore blocks before this number